### PR TITLE
fix delegate selection set pruning to use parent types/parent type fi…

### PR DESCRIPTION
**Pull request by brianjquinn**
…elds rather than getFieldDef()

Through some testing - I found that getFieldDef() was not the correct method to use to determine if a field in a delegated selection set is present/can be resolved by the schema being delegated to.

### CODE REVIEW

File: lib/delegate/index.js [hunk 1]
```diff
@@ -180,14 +184,16 @@ const createSubOperationDocument = function (component, targetRootField, args, s
...
-  // in the schema we are delegating to
-  visitFunctions[Kind.FIELD] = function () {
-    const fieldDef = typeInfo.getFieldDef();
-    // if the field isn't defined in delegated schema, return null to remove the field from the delegated document
-    if (!fieldDef) {
-      return null;
+  // prune selection set fields that are not defined in the target schema
+  visitFunctions[Kind.FIELD] = function (node) {
+    const parentType = typeInfo.getParentType();
+    if (isObjectType(parentType) || isInterfaceType(parentType)) {
```
Comment (roymanas) [inline lib/delegate/index.js:190]: Will this check be isAbstractType() instead of just isInterfaceType() to cover both Union and Interfaces?

**Review by roymanas — COMMENTED** [review]
tests for delegate scenarios were passing using the getFieldDef() way, though actually it should be getParentType(). Do we need to add more test cases to cover edge cases with delegateToComponent’s pruning functionality?

**Comment by brianjquinn** [comment]
>Do we need to add more test cases to cover edge cases with delegateToComponent’s pruning functionality?

I'll add one more test case for delegating from a non-root resolver with pruning, since we have a pruning test case for delegating from a root resolver.

File: lib/delegate/index.js (hunk 1 above)
Comment (brianjquinn) [reply lib/delegate/index.js:190]: nice catch - I'll change this

**Review by roymanas — APPROVED** [review]
